### PR TITLE
Rubocop integration

### PR DESCRIPTION
**Pull request by linkyndy**
I (tried to) integrate Rubocop here through our gem. Fixed some of the offenses, but there are still some left that you're more entitled to decide upon. Let me know if something is not working for you :)

### CODE REVIEW

File: lib/zip_tricks/file_reader.rb [hunk 1]
```diff
@@ -669,6 +669,7 @@ def num_files_and_central_directory_offset_zip64(io, zip64_end_of_cdir_location)
   # central directory record upfront (all the fixed fields + at most 0xFFFF
   # bytes of the archive comment)
   MAX_END_OF_CENTRAL_DIRECTORY_RECORD_SIZE =
+    # rubocop:disable Layout/MultilineOperationIndentation
```
Comment (julik) [inline lib/zip_tricks/file_reader.rb]: What is the proposed Rubocop formatting for these constants defined with a block?

**Comment by julik** [comment]
❗️ BTW what's the idea of approving with CI failing still? That's not how it works :-) that's not how anything of this works....

File: lib/zip_tricks/file_reader.rb (hunk 1 above)
Comment (linkyndy) [reply lib/zip_tricks/file_reader.rb]: Well:

```
1+
2+
...
```

So, no indentation.

**Comment by linkyndy** [comment]
@julik I wasn't seeking approval yet, but more your contribution to fixing the offenses, since you know best what's supposed to be there (see my PR description) 😀

**Comment by julik** [comment]
Ah, so that's how them tables turn now. Ok ok, but I won't get to this until the end of this week I am afraid :-(

**Comment by julik** [comment]
@linkyndy I think that will do. I have applied 3 exceptions in total:

* The QA scripts generate large files and are messy, they are also not used anywhere else - I think using global variables in them is justified
* I absolutely hate the first-argument-on-next-line rules. Can't help it. Can we maybe reconsider them? This is a personal preference.

The rest should be appeased.

**Comment by linkyndy** [comment]
Great job! I agree some projects can have exceptions from the default cops list, so that's fine. Regarding the line breaks/first arguments, can you point me to an example? I didn't see one while I was skimming through the changes.

File: .rubocop.yml [hunk 2]
```diff
@@ -1,141 +1,10 @@
...
-
-Lint/AssignmentInCondition:
+inherit_gem:
+  wetransfer_style: ruby/default.yml
+Layout/FirstMethodArgumentLineBreak:
   Enabled: false
-
-# We use % all over and nothing ever came out of it
-Style/FormatString:
+Layout/FirstMethodParameterLineBreak:
```
Comment (davidenko87) [inline .rubocop.yml]: can we add these two to the general style gem?

Comment (linkyndy) [reply .rubocop.yml]: Can you point me to an actual example of this in the PR? Not that it is dependent on including it in the style gem, but I want to understand better what's the problem.

Comment (davidenko87) [reply .rubocop.yml]: For example, in `size_estimator.rb` and `steamer.rb` we place the arguments on the next line. Inside `size_estimator_spec.rb` and `file_reader_spec.rb` these are on one line or the first line is in the same as the method.

Comment (linkyndy) [reply .rubocop.yml]: Well, I think the problem is occurring only for `size_estimator_spec.rb`, where I tend to agree with rubocop. That way of aligning arguments is very specific to Python. But anyways, that's my taste, so if you're against it, let's remove the cops from `wetransfer_style`. The aim is to enforce stuff while keeping everyone happy (within reasonable boundaries) :)

**Comment by julik** [comment]
@davidenko87 fair point

**Comment by linkyndy** [comment]
So what do you guys propose?

**Comment by linkyndy** [comment]
Well, removing the cops will allow multiple styles, so it cancels your first statement. How do you want to enforce that style?

**Comment by davidenko87** [comment]
Well, we could edit the linter so it matches our needs.

**Comment by linkyndy** [comment]
Wouldn't go that far...

**Comment by davidenko87** [comment]
What is your proposal to have one style in this case?

**Comment by linkyndy** [comment]
Either keep the cops, either remove them and allow people different styles. We are anyway disagreeing on which style to use. Maybe it's not that bad to be a bit flexible and allow everyone the freedom to use whatever style they want.

**Comment by julik** [comment]
@linkyndy @davidenko87 I think we are in the clear now. I reverted the changes for first-arg-on-next-line to have consistent style and applied a reformat so that args/kwargs get realigned. I think this can be merged 🎉

**Comment by davidenko87** [comment]
i agree!

**Comment by julik** [comment]
There is a couple exceptions that I missed when I was removing reminder comments, I'll deal with them.

**Comment by julik** [comment]
@linkyndy aaaalmost there. Can you take a look how we should deal with codeclimate? If it is going to keep failing like this we should remove it since we will be running rubocop for each CI cycle anyway.

**Comment by julik** [comment]
Killed Codeclimate for now, we can revisit later ;-)

**Comment by linkyndy** [comment]
Yep, looks good! Regarding Codeclimate, never actually opened it 😛 But indeed, we should rethink so it doesn't stay in our way. I like it as a tool, so would be nice to manage to keep it somehow.

**Pull request merged**